bundles/org.openhab.binding.miio/src/main/java/org/openhab/binding/miio/internal/robot/RRMapFileParser.java

```diff
@@ -0,0 +1,415 @@
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.miio.internal.robot;
+
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.zip.GZIPInputStream;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.openhab.binding.miio.internal.Utils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link RRMapFileParser} is used to parse the RR map file format created by Xiaomi / RockRobo vacuum
+ *
+ * @author Marcel Verpaalen - Initial contribution
+ */
+@NonNullByDefault
+public class RRMapFileParser {
+    public static final int CHARGER = 1;
+    public static final int IMAGE = 2;
+    public static final int PATH = 3;
+    public static final int GOTO_PATH = 4;
+    public static final int GOTO_PREDICTED_PATH = 5;
+    public static final int CURRENTLY_CLEANED_ZONES = 6;
+    public static final int GOTO_TARGET = 7;
+    public static final int ROBOT_POSITION = 8;
+    public static final int NO_GO_AREAS = 9;
+    public static final int VIRTUAL_WALLS = 10;
+    public static final int BLOCKS = 11;
+    public static final int MFBZS_AREA = 12;
+    public static final int OBSTACLES = 13;
+    public static final int DIGEST = 1024;
+    public static final int HEADER = 0x7272;
+
+    public static final String PATH_POINT_LENGTH = "pointLength";
+    public static final String PATH_POINT_SIZE = "pointSize";
+    public static final String PATH_ANGLE = "angle";
+
+    private static final float MM = 50.0f;
+
+    private byte[] image = new byte[] { 0 };
+    private int majorVersion;
+    private int minorVersion;
+    private int mapIndex;
+    private int mapSequence;
+    private boolean isValid;
+
+    private int imgHeight;
+    private int imgWidth;
+    private int imageSize;
+    private int top;
+    private int left;
+    private int offset;
+
+    private int chargerX;
+    private int chargerY;
+    private int roboX;
+    private int roboY;
+    private int roboA;
+    private float gotoX = 0;
+    private float gotoY = 0;
+    private Map<Integer, ArrayList<Float[]>> paths = new HashMap<>();
+    private Map<Integer, Map<String, Integer>> pathsDetails = new HashMap<>();
+    private Map<Integer, ArrayList<Float[]>> areas = new HashMap<>();
+    private ArrayList<Float[]> walls = new ArrayList<Float[]>();
+    private ArrayList<Float[]> zones = new ArrayList<Float[]>();
```

Comment:
Why `Float[]` instead of `float[]`?
```suggestion
    private ArrayList<Float[]> walls = new ArrayList<>();
    private ArrayList<Float[]> zones = new ArrayList<>();
```